File: Day-02/python/rainbowdashlabs/day_2.py
first = {
    "A": "Rock",
    "B": "Paper",
    "C": "Scissor"
}

second = {
    "X": "Rock",
    "Y": "Paper",
    "Z": "Scissor"
}

points = {
    "Rock": 1,
    "Paper": 2,
    "Scissor": 3
}

choices = [e.strip().split(" ") for e in open('input.txt')]

print(choices)
ico = [[first[e[0]], second[e[1]]] for e in choices]


def check(other, my) -> int:
    point = points[my]

    if other == my:
        return 3 + point

    if other == "Scissor" and my == "Paper":
        return 0 + point
    if other == "Scissor" and my == "Rock":
        return 6 + point

    if other == "Paper" and my == "Rock":
        return 0 + point

    if other == "Paper" and my == "Scissor":
        return 6 + point

    if other == "Rock" and my == "Paper":
        return 6 + point

    if other == "Rock" and my == "Scissor":
        return 0 + point


print([check(*e) for e in ico])

print(f"Part 1 {sum([check(*e) for e in ico])}")

win = {
    "Scissor": "Rock",  # Loose
    "Paper": "Scissor",  # draw
    "Rock": "Paper"  # Win
}

loose = {
    "Rock": "Scissor",  # Loose
    "Scissor": "Paper",  # draw
    "Paper": "Rock"  # Win
}


def win_con(other, my):
    if my == "X":
        return [other,loose[other]]

    if my == "Y":
        return [other,other]

    if my == "Z":
        return [other, win[other]]

print(f"Part 1 {sum([check(*win_con(first[e[0]], e[1])) for e in choices])}")

File: Day-02/rust/Marie/day2.rs
use std::fs;

fn main() {
    let input = fs::read_to_string("inputs/day2.txt").unwrap();
    let total: i32 = input
        .lines()
        .map(|line| {
            let mut splitted = line.split(' ');
            let first = splitted.next().unwrap();
            let second = splitted.next().unwrap();

            let first_score = match first.chars().next().unwrap() {
                'A' => 1, // Rock
                'B' => 2, // Paper
                'C' => 3, // Scissors
                _ => panic!(),
            };
            let second_score = match second.chars().next().unwrap() {
                'X' => 1, // Rock
                'Y' => 2, // Paper
                'Z' => 3, // Scissors
                _ => panic!(),
            };

            match first_score - second_score {
                -1 | 2 => second_score + 6,
                1 | -2 => second_score,
                0 => second_score + 3,
                _ => panic!(),
            }
        })
        .sum();
    println!("{total}");

    let total: i32 = input
        .lines()
        .map(|line| {
            let mut splitted = line.split(' ');
            let first = splitted.next().unwrap();
            let second = splitted.next().unwrap();

            let first_score = match first.chars().next().unwrap() {
                'A' => 1, // Rock
                'B' => 2, // Paper
                'C' => 3, // Scissors
                _ => panic!(),
            };
            match second.chars().next().unwrap() {
                'X' => match first_score {
                    1 => 3,
                    i => i - 1,
                }, // Loose
                'Y' => first_score + 3, // Draw
                'Z' => {
                    let score = match first_score {
                        3 => 1,
                        i => i + 1,
                    };
                    score + 6
                } // Win
                _ => panic!(),
            }
        })
        .sum();
    println!("{total}");
}
